File: 7revertInt/main.php
<?php

class Solution {

    /**
     * @param Integer $x
     * @return Integer
     */
    function reverse($x) {
        $n = 0;
        while ($x != 0) {
            $n *= 10;
            $n += $x % 10;
            $x = ($x - ($x%10))/10;
        }
        if ($n > 2147483647 || $n < -2147483648) {
            return 0;
        }
        return $n;
    }
}

echo (new Solution())->reverse(-123);

File: 7revertInt/main.go
package main

import "fmt"

func reverse(x int) int {
	n := 0
	for x != 0 {
		n *= 10
		n += x % 10
		x = x / 10
	}
	if n > 2147483647 || n < -2147483648 {
		return 0
	}
	return n
}

func main() {
	x := reverse(-123456789)
	fmt.Printf("%d\n", x)
}
